vra/resource_project.go

```diff
@@ -321,7 +321,11 @@ func flattenZoneAssignment(list []*models.ZoneAssignment) []map[string]interface
 }
 
 func expandProjectConstraints(configProjectConstraints []interface{}) map[string][]models.Constraint {
-	projectConstraints := make(map[string][]models.Constraint)
+	projectConstraints := map[string][]models.Constraint{
+		"extensibility": {},
```

Comment:
Does this mean we are adding these additional fields to the Constraint model generated from the swagger?

Reply:
We were already doing that: 

https://github.com/vmware/terraform-provider-vra/blob/2f6d4c547974ff481f22898b5654dd1243dfb27e/vra/resource_project.go#L330 

Here what we are doing is to initialize the list of constraints with an empty key (`extensibility`, `network`, `storage`), so when we do a `PATCH`, we're effectively removing any existing constraints if they are not set at the terraform manifest file. 

Regarding the model, the swagger does NOT contain those fields, it works by convention (the backend will recognize those keys): 

https://github.com/vmware/vra-sdk-go/blob/db1e1fa0c66c9578b2e5a3628a5bd6da0fd39c4c/swagger/vra-iaas.json#L12190